File: h5/src/api/feedback.ts
import { get, post } from './index'
import type { Feedback } from './types'

/**
 * 反馈评价 API(员工自助)
 * 后端:FeedbackController
 * - POST /feedback:员工创建反馈
 * - GET /feedback/my:员工查询自己的反馈列表(非分页)
 * - GET /feedback/my/{id}:员工查询反馈详情
 */

/** 创建反馈/评价(员工可调用) */
export function createFeedback(payload: {
  storeId: number
  employeeId: number
  orderId?: number | null
  dishId?: number | null
  rating: number
  content?: string
  category?: number
}): Promise<Feedback> {
  return post<Feedback>('/feedback', payload)
}

/** 查询我的反馈列表(非分页,后端返回全部) */
export function getMyFeedback(): Promise<Feedback[]> {
  return get<Feedback[]>('/feedback/my')
}

/** 查询反馈详情 */
export function getFeedbackDetail(id: number): Promise<Feedback> {
  return get<Feedback>(`/feedback/my/${id}`)
}

File: backend/src/main/java/com/example/canteen/entity/Feedback.java
package com.example.canteen.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("feedback")
public class Feedback {
    @TableId(type = IdType.AUTO)
    private Long id;
    private Long storeId;
    private Long employeeId;
    /** 关联订单(可为空) */
    private Long orderId;
    /** 关联菜品(可为空) */
    private Long dishId;
    /** 评分 1-5 */
    private Integer rating;
    /** 反馈内容 */
    private String content;
    /** 1=菜品评价 2=服务投诉 3=建议 4=其他 */
    private Integer category;
    /** 1=待处理 2=已处理 3=已忽略 */
    private Integer status;
    /** 管理员回复 */
    private String reply;
    /** 回复人 */
    private Long replyAdminId;
    private LocalDateTime repliedAt;
    private LocalDateTime createdAt;

    @TableField(exist = false)
    private String employeeName;
    @TableField(exist = false)
    private String dishName;
}
